tweakwcs/linearfit.py

```diff
@@ -488,13 +507,14 @@ def fit_rscale(xy, uv, wxy=None, wuv=None):
         u -= um
         v -= vm
 
-        su2 = np.dot(w, u**2)
-        sv2 = np.dot(w, v**2)
         sxv = np.dot(w, x * v)
         syu = np.dot(w, y * u)
         sxu = np.dot(w, x * u)
         syv = np.dot(w, y * v)
-        su2v2 = su2 + sv2
+        if scale is None:
+            su2 = np.dot(w, u**2)
+            sv2 = np.dot(w, v**2)
+            su2v2 = su2 + sv2
 
```

Comment:
I have a sense of deja vu.

Reply:
> I have a sense of deja vu.

What do you mean?

Reply:
My bad. I didn't look carefully enough in doing the comparison.